scheduler/scheduler.go

```diff
@@ -151,30 +151,36 @@ func spawnHosts(ctx context.Context, newHostsNeeded map[string]int) (map[string]
 		}
 
 		hostsSpawnedPerDistro[distroId] = make([]host.Host, 0, numHostsToSpawn)
-		// if distro can have containers, check if there are enough parents to hold
-		// new containers
-		if d.MaxContainers > 0 {
-			currentParents, err := host.FindAllRunningParentsByDistro(distroId)
+
+		// if distro is container distro, check if there are enough parent hosts to
+		// support new containers
+		if pool != (evergreen.ContainerPool{}) {
+
+			// find all running parents with the specified container pool
+			currentParents, err := host.FindAllRunningParentsByContainerPool(pool.Id)
```

Comment:
Didn't we decide that this would only run for the distros of the containers themselves? Since parent distros won't have tasks, I'm not sure this will work